src/Features/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -120,16 +120,6 @@ public ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities)
             // Using VS server capabilities because we have our own custom client.
             capabilities.OnAutoInsertProvider = new VSInternalDocumentOnAutoInsertOptions { TriggerCharacters = ["'", "/", "\n"] };
 
-            if (!supportsVsExtensions)
-            {
-                capabilities.DiagnosticOptions = new DiagnosticOptions
-                {
-                    InterFileDependencies = true,
-                    WorkDoneProgress = true,
-                    WorkspaceDiagnostics = true,
-                };
-            }
```

Comment:
what happened to this?

Reply:
looks like a bad merge

Reply:
I believe this is intentional - since we're dynamically registering sources we do not need to statically register for diagnostics.